### ISSUE TITLE
config.cmaq.csh

### ISSUE THREAD

**Issue by zacadelman** [USEPA/CMAQ#47]
A few changes to make this script work for the general user. The diffs below are in this form: diff cmas.config.csh epa.config.csh

Add a link flag for Intel compilers. This is a generic flag that the I/O API uses and it needs to be in there for the programs that use Intel-compiled I/O API libraries.
```
94c94
<         setenv myLINK_FLAG "-openmp"
---
>         setenv myLINK_FLAG "
```

You are linking the netcdf library directly into the NETCDF_DIR. This doesn't work because bldmake is looking in NETCDF_DIR/lib for the library.  Either fix this in bldmake, or just make the change that I suggest here to the config.cmaq.csh script. 
```
190,191c189
<  if ( ! -d $NETCDF_DIR )  mkdir $NETCDF_DIR
<     ln -s $NETCDF_LIB_DIR $NETCDF_DIR/lib
---
>  if ( ! -d $NETCDF_DIR )  ln -s $NETCDF_LIB_DIR $NETCDF_DIR
```
We can get rid of the check for the NetCDF includes, as they're not needed anymore:
```
204,207c202,205
< # if ( ! -e $NETCDF_DIR/include/netcdf.h ) then 
< #    echo "ERROR: $NETCDF_DIR/include/netcdf.h does not exist in your CMAQ_LIB directory !!! Check your installation before proceeding with CMAQ build."
< #    exit
< # endif
---
>  if ( ! -e $NETCDF_DIR/include/netcdf.h ) then 
>     echo "ERROR: $NETCDF_DIR/include/netcdf.h does not exist in your CMAQ_LIB directory !!! Check your installation before proceeding with CMAQ build."
>     exit
>  endif
```

**Comment by zacadelman** [comment]
Change myFC from mpiifort to mpif90 as the default.  This change is particularly needed for the PGI and GCC compile settings.  I think mpif90 is the more generic MPI wrapper.

**Comment by zacadelman** [comment]
Change the MPI_LIB_DIR to point to the more generic top level MPI directory that contains both the lib and include files (rather than pointing directly to lib).  I found that with our PGI mpi compile, it was looking for mpi.h, for some reason not picking it up from the mpif90 wrapper.

**Comment by bnmurphy** [comment]
So, do get rid of netcdf include folder and links?
And do NOT get rid of mpi libraries?

**Comment by zacadelman** [comment]
I'm not sure about the MPI libraries anymore now.  Our PGI wrapper (mpif90) did not pull in the include file required by dist_envr.c. The compile actually need the path to the mpi.h to work.  I didn't see this with the Gfortran and Intel compilers.  I'm going to look at this in more detail after the PGI run finishes in a few minutes.  

So yes on the netcdf include folder and links.  And wait on the MPI include folder and links.  I'll report back when I know more.

**Comment by zacadelman** [comment]
Removed the -mcmodel=medium compiler/linker flag for Portland Group.  Information from Carlie on this option (note that it used to be in the scripts for the Intel compiler, hence Carlie's note below), but now it's only there for PGI:

```
This is used when either stack > 2GB or individual arrays > 2 GB
are needed.  All of the code (libraries included) must be compiled
this way; "lib*.a" static libraries are not supported ("lib*.so"
shared libraries only).  This is NOT needed for most modeling tasks;
medium-model libraries and object files MUST be kept separate from
normal ("-mcmodel=small") libraries and object files.  See:

<https://stackoverflow.com/questions/13936758/intel-fortran-compiler-with-large-file-size>

   <https://www-01.ibm.com/support/docview.wss?uid=isg3T1014413>

   [Intel seems to be revising their web-site; the links and
    web-searches for "mcmodel" go to "page not found" stuff...]

Note that the Intel-compiler binary type for medium memory model
is Linux2_x86_64ifort_medium; we don't have a corresponding type
for GNU or PG (but one would be easy to construct; note that for
the GNU compilers, this also needs that the compiler have been
built with "-mcmodel=medium", else it will silently build 32-bit
object files).
```

**Comment by bnmurphy** [comment]
Ok, these have been addressed and are waiting internal approval before pushing to the public repo
